design/authenticators/authn_jwt/token_schema.md

```diff
@@ -0,0 +1,178 @@
+
+## Glossary
+
+[//]: # "Describe terms that will be used throughout the design"
+[//]: # "You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#"
+
+| **Term**         | **Description**                                              |
+| ---------------- | ------------------------------------------------------------ |
+| JWT Claims       | Claims are pieces of information asserted about a subject    |
+| Host Annotation  | Defines the checks that we should do on authentication request for host. In JWT authentication its a claim check. |
+| Mandatory Claims | JWT-specific claims that a host must validate their existance and value |
+| Claims Mapping   | Mapping between claim name and name of host annotation       |
+
+## Useful Links
+[//]: # "Add links that may be useful for the reader"
+
+| **Name**   | **Link**                                                     |
+| ---------- | ------------------------------------------------------------ |
+| JWT Claims | https://auth0.com/docs/tokens/json-web-tokens/json-web-token-claims |
+
+## Requirements
+[//]: # "Elaborate on the issue you are writing a solution for"
+
+1. Required fields
+   1. In the authenticator policy, the user will have the ability to declare which claims must be in host annotations. As a result, also in the JWT token
+   2. When sending an authentication request, the authenticator MUST validate those claims - existence and content 
+2. Mapping 
+   1. The user should have the ability to map a given "technical" claim name to a more familiar / business one - i.e, in git-lab. Instead of mentioning in the policy "sub" claim, the user will have the ability to mention "job"
+
+## Solution
+User will define the token schema variables in the authenticator policy. The variables will be checked in the JWT authentication request.
+
+* The usage of this variables is optional and non mandatory. User can authenticate to JWT in the same way he authenticated in stage 1. Using the feature adds more security checks to the authentication.
+
+### User Interface
+
+#### Authenticator Policy
+
+Two new variables will be added to the authenticator policy:
+
+* mandatory_claims
+* claims_mapping
+
+```yaml
+    - !policy
+    id: conjur/authn-jwt/VendorY
+    body:
+    - !webservice
+    
+    - !variable
+      id: mandatory_claims
+      
+    - !variable
+      id: claims_mapping
+```
+
+#### Variable Values Example
+
+| Name             | Value                   | Description                                      |
+| ---------------- | ----------------------- | ------------------------------------------------ |
+| mandatory_claims | branch, job, aud        | List of claims as they appear in host annotation |
+| claims_mapping   | {ref: branch, sub: job} | Mapping between host annotation to JWT claim     |
+
+* For any claim mapped, we will add the host annotation to the `mandatory_claims` variable and not the JWT claim name.
```

Comment:
I think that "use" will be better here.. we will not add it.. the user adds it.. we will use the mapped name instead

Reply:
?